application/views/collection_instrument_view.py

```diff
@@ -46,9 +46,34 @@ def collection_instrument_by_search_string():
 @auth.login_required
 @collection_instrument_view.route('/collectioninstrument/id/<instrument_id>', methods=['GET'])
 def collection_instrument_by_id(instrument_id):
-    instrument_json = CollectionInstrument().get_instrument_json_by_id(instrument_id)
+    instrument_json = CollectionInstrument().get_instrument_json(instrument_id)
 
     if instrument_json:
         return make_response(jsonify(instrument_json), 200)
     else:
         return make_response(COLLECTION_INSTRUMENT_NOT_FOUND, 404)
+
+
+@auth.login_required
+@collection_instrument_view.route('/download/<instrument_id>', methods=['GET'])
+def instrument_data(instrument_id):
+    data, ru_ref = CollectionInstrument().get_instrument_data(instrument_id)
+
+    if data and ru_ref:
+        response = make_response(data, 200)
+        response.headers["Content-Disposition"] = "attachment; filename={}.xlsx".format(ru_ref)
+        response.headers["Content-type"] = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+    else:
+        response = make_response(COLLECTION_INSTRUMENT_NOT_FOUND, 404)
+    return response
+
+
+@auth.login_required
+@collection_instrument_view.route('/instrument_size/<instrument_id>', methods=['GET'])
+def instrument_size(instrument_id):
+    instrument = CollectionInstrument().get_instrument_json(instrument_id)
+
+    if instrument and 'len' in instrument:
+        return make_response(jsonify(instrument['len']), 200)
+    else:
```

Comment:
Since we return in the if statement the else here is unnecessary. 
It might be cleaner to do 
```
if not instrument...:
    return (..., 404)
return make_response(jsonify(instrument['len']), 200)
```

Reply:
Fair comment, will update